File: src/app/model/LongitudeMetric.ts
import { LocationServiceEvent } from './LocationServiceEvent';
import { MetricService } from "../services/metric.service";
import { Metric } from "./Metric";

export class LongitudeMetric extends Metric {
    private handler = this.locationEventHandler.bind(this);

    constructor(metricService: MetricService) {
        super('Longitude', '°', metricService, '°', 5);
    }

    startLogging(): void {
        this.metricService.getLocationService().subscribeForLocation(this.handler);
    }

    stopLogging(): void {
        this.metricService.getLocationService().unsubscribeForLocation(this.handler);
    }

    locationEventHandler(event: LocationServiceEvent): void {
        this.addValue(event.location.coords.longitude, new Date(event.location.timestamp));
    }
}

File: android/app/src/main/java/com/openbikecomputer/model/LongitudeMetric.java
package com.openbikecomputer.model;

import com.openbikecomputer.service.IMetricService;
import com.openbikecomputer.service.OnLocationUpdate;

public class LongitudeMetric extends Metric implements OnLocationUpdate {
    public LongitudeMetric(IMetricService metricService) {
        super(MetricType.LONGITUDE, "Longitude", "\u00b0", metricService, "\u00b0", 5);
    }

    @Override
    public void startLogging() {
        metricService.getLocationService().subscribeForLocation(this);
    }

    @Override
    public void stopLogging() {
        metricService.getLocationService().unsubscribeForLocation(this);
    }

    @Override
    public void onLocation(LocationSample sample) {
        addValue(sample.longitude, sample.timestamp);
    }
}
